server/polar/customer_portal/endpoints/subscription.py

```diff
@@ -104,6 +105,93 @@ async def get(
     return subscription
 
 
+class SubscriptionChargePreviewResponse(TypedDict):
+    base_amount: int
+    metered_amount: int
+    subtotal_amount: int
+    discount_amount: int
+    tax_amount: int
+    total_amount: int
+
+
+@router.get(
+    "/{id}/charge-preview",
+    summary="Preview Next Charge For Active Subscription",
+    response_model=SubscriptionChargePreviewResponse,
+    responses={404: SubscriptionNotFound},
+    tags=[APITag.private],
+)
+async def get_charge_preview(
+    id: SubscriptionID,
+    auth_subject: auth.CustomerPortalRead,
+    session: AsyncSession = Depends(get_db_session),
+) -> SubscriptionChargePreviewResponse:
+    """Get current period usage and cost breakdown for a subscription."""
+    subscription = await customer_subscription_service.get_by_id(
+        session, auth_subject, id
+    )
+
+    if subscription is None:
+        raise ResourceNotFound()
+
+    if subscription.status != "active":
+        ## FIXME Is a 404 the correct behavior?
+        raise ResourceNotFound()
+
+    base_price = sum(p.amount for p in subscription.subscription_product_prices)
+
+    metered_amount = sum(meter.amount for meter in subscription.meters)
+
+    subtotal_amount = base_price + metered_amount
+
+    discount_amount = 0
+
+    # Ensure the discount has not expired yet
+    if subscription.discount is not None:
+        assert subscription.started_at is not None
+        if subscription.discount.is_repetition_expired(
+            subscription.started_at, subscription.current_period_start
```

Comment:
Since we're looking at the _next_ invoice we should check if the discount expires on the next period. So it should rather be:

```py
subscription.discount.is_repetition_expired(subscription.started_at, subscription.current_period_end)
```